src/emqtt_frame.erl

```diff
@@ -336,6 +336,9 @@ parse_properties(Bin, ?MQTT_PROTO_V5) ->
     <<PropsBin:Len/binary, Rest1/binary>> = Rest,
     {parse_property(PropsBin, #{}), Rest1}.
 
+parse_property(<<>>, Props = #{'Subscription-Identifier' := Ids})
+  when is_list(Ids) ->
+    maps:update('Subscription-Identifier', lists:reverse(Ids), Props);
```

Comment:
maybe we don't need reverse here?

Reply:
You're right, we don't need the `lists:reverse/1` since the order of subscription identifiers is not important according to the spec.
Advantages of the `lists:reverse/1` here are:
1. The client gets the same order of subscription identifiers as sent by the server. This may help understandability and debugging. But again, order doesn't matter.
2. The symmetric property test added by this PR ensures that serialising and parsing a packet results in the same packet. Since serialise happens in the correct order, for the test to succeed, parsing also needs to happen in the correct order. But I can re-write that test.

So, I'm happy to remove the `lists:reverse/1` and re-write the test. However, I don't think that the list reversal is a performance issue here. A client receiving a PUBLISH packet with many subscription identifiers is an edge case.